### PR TITLE
[MM-48727] Restore license checks for SSO in self-hosted deployments - admin console

### PR DESCRIPTION
**Pull request by plant99**
#### Summary

Please see description in [JIRA ticket](https://mattermost.atlassian.net/browse/MM-48727).
#### Ticket Link

[JIRA ticket](https://mattermost.atlassian.net/browse/MM-48727)
#### Related Pull Requests

- Has server changes ([please link here](https://github.com/mattermost/mattermost-server/pull/21781))
- Has mobile changes (please link here)

#### Release Note

```release-note
NONE
```

**QA Steps**

Please see QA steps in [JIRA ticket](https://mattermost.atlassian.net/browse/MM-48727),

### CODE REVIEW

File: components/admin_console/admin_definition.jsx [hunk 1]
```diff
@@ -4795,7 +4797,14 @@ const AdminDefinition = {
             title: t('admin.sidebar.oauth'),
             title_default: 'OAuth 2.0',
             isHidden: it.any(
-                it.not(usesLegacyOauth),
+                it.any(
+                    it.not(it.licensed),
+                    it.all(it.licensedForSku('starter')),
```
Comment (mkraft) [inline components/admin_console/admin_definition.jsx:4802]: Can these be simplified to the following?
```suggestion
                    it.licensedForSku('starter'),
```

File: components/admin_console/admin_definition.jsx [hunk 2]
```diff
@@ -4795,7 +4797,14 @@ const AdminDefinition = {
...
+                ),
+                it.all(
+                    it.all(it.licensedForFeature('OpenId')),
```
Comment (mkraft) [inline components/admin_console/admin_definition.jsx:4805]: and this one?
```suggestion
                    it.licensedForFeature('OpenId'),
```

**Review by mkraft — APPROVED** [review]

**Review by BenCookie95 — APPROVED** [review]

**Review by furqanmlk — APPROVED** [review]
Working fine

**Pull request merged**